src/bbc_meet_spotify/spotify.py

```diff
@@ -29,10 +29,11 @@ def add_songs_to_playlist(self, playlist_id: str, song_ids: List[str]):
         playlist_info = self.spotify.user_playlist(self.username, playlist_id, "tracks")
         existing_songs = [x["track"]["id"] for x in playlist_info["tracks"]["items"]]
         new_song_ids = [song_id for song_id in song_ids if song_id not in existing_songs]
-        try:
+        if new_song_ids:
             self.spotify.user_playlist_add_tracks(self.username, playlist_id, new_song_ids)
-        except SpotifyException as e:
```

Comment:
The exception should now bubble up and exit, without writing the playlist history